spec/controllers/saml_idp_controller_spec.rb

```diff
@@ -214,10 +214,21 @@
         stub_analytics
         allow(@analytics).to receive(:track_event)
 
-        saml_get_auth(missing_authn_context_saml_settings)
+        user = create(:user, :signed_up)
+        generate_saml_response(user, missing_authn_context_saml_settings)
+
+        expect(response.status).to eq(200)
```

Comment:
I don't disagree with you. I was just trying to keep it consistent with all the specs around it. They all have the format

* set up stuff
* generate response
* check response.status
* build analytics hash
* check analytics